### PR TITLE
Alerting: Enable simplified routing FF by default

### PR DESCRIPTION
**Pull request by JacobsonMT**
**What is this feature?**

Moves simplified routing feature flag `alertingSimplifiedRouting` to GA and enables by default. Removes admonitions from docs about the FF.

**Why do we need this feature?**

Simplified routing should be GA and enabled by default in G11.

**Who is this feature for?**

Grafana Alerting users.

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 1]
```diff
@@ -2067,6 +2078,23 @@ func TestIntegrationAlertmanagerStatus(t *testing.T) {
 	}
 }
 `
+	cfgWithoutAutogen := fmt.Sprintf(cfgTemplate, `{
```
Comment (yuri-tceretian) [inline pkg/tests/api/alerting/api_alertmanager_test.go:2081]: perhaps move it to another PR?

**Review by yuri-tceretian — APPROVED** [review]

**Comment by brendamuir** [comment]
Hey Matt -- so do we only need to keep this section in the docs then? https://grafana.com/docs/grafana-cloud/alerting-and-irm/alerting/alerting-rules/create-grafana-managed-rule/#configure-labels-and-notifications and can i delete this section (that describes the process when simplified is NOT enabled): https://grafana.com/docs/grafana-cloud/alerting-and-irm/alerting/alerting-rules/create-grafana-managed-rule/#configure-notifications ? if so, lets also delete that section with this PR :)

**Comment by JacobsonMT** [comment]
> Hey Matt -- so do we only need to keep this section in the docs then? https://grafana.com/docs/grafana-cloud/alerting-and-irm/alerting/alerting-rules/create-grafana-managed-rule/#configure-labels-and-notifications and can i delete this section (that describes the process when simplified is NOT enabled): https://grafana.com/docs/grafana-cloud/alerting-and-irm/alerting/alerting-rules/create-grafana-managed-rule/#configure-notifications ? if so, lets also delete that section with this PR :)

Ah, nice catch! Didn't realize the section was duplicated. Yes, I'll make the change in this PR.

File: pkg/tests/api/alerting/api_alertmanager_test.go (hunk 1 above)
Comment (JacobsonMT) [reply pkg/tests/api/alerting/api_alertmanager_test.go:2081]: The test will fail without this change, I could perhaps disable the simplfied routing FF here instead but fixing the test seems best.

**Review by brendamuir — APPROVED** [review]
